### ISSUE TITLE
Changing query params empties a hash

### ISSUE THREAD

**Issue by RomanTokar** [pbeshai/use-query-params#229]
That's likely because we just only pass location.search in `react-router`'s history.replace() or history.push() method instead of the entire location object.
```javascript
// @ts-ignore
import { useHistory, useLocation } from 'react-router';
import {
  QueryParamAdapter,
  QueryParamAdapterComponent,
} from 'use-query-params';

/**
 * Query Param Adapter for react-router v5
 */
export const ReactRouter5Adapter: QueryParamAdapterComponent = ({
  children,
}) => {
  const location = useLocation();
  const history = useHistory();

  const adapter: QueryParamAdapter = {
    replace(location) {
      history.replace(location.search || '?', location.state);
    },
    push(location) {
      history.push(location.search || '?', location.state);
    },
    get location() {
      return location;
    },
  };

  return children(adapter);
};
```

**Comment by pbeshai** [comment]
Yeah that's very true. I made this change based on the way react-router works with search params (see [this comment](https://github.com/remix-run/react-router/issues/8393#issuecomment-979676684) for their explanation on removing the hash). I don't necessarily agree with their reasoning, but I don't really use hashes myself very much to have a strong opinion.  I'm unsure whether or not the provided adapter should do as you show above or not.

**Comment by RomanTokar** [comment]
I guess I also don't agree.
Here is an [codesandbox](https://codesandbox.io/s/use-query-params-issue-2eh8n7?file=/src/index.js) with applied above adapter.

**Comment by andymerskin** [comment]
@roman

Great fix, thanks for sharing! Totally agree that the hash shouldn't get nuked when search param changes happen. A lot of applications have tab components or other interactive UI elements that use URL hashes to target them, like going directly to a route with a specific tab selected when the page loads.

For others needing to use this fix in their project locally:
1. Modify `./node_modules/use-query-params/adapters/react-router-5/index.js` in your project with the above changes
2. Use `patch-package` [link]() to store a diff of the changes. This yells your package manager to install your local patch whenever you install dependencies for your project, and helps document changes you've made to packages. Useful for bugfixes like these when your favorite package has lost maintainers or isn't moving on fixes.
3. Import the JS file directly, rather than the built `index.cjs.js` file by default:
    ```typescript
    import { ReactRouter5Adapter } from "use-query-params/adapters/react-router-5/index";
    ```
    
    Your project may need `allowJs` in your TypeScript config. Change your configuration accordingly. The import statement may vary depending on your bundling and build setup, but you just need to import the unbuilt JS file from source that you've modified.

**Comment by vincerubinetti** [comment]
Having the same issue with `ReactRouter6Adapter`. Surprised this isn't a more common complaint and urgent fix. I feel like this library should try to be as non-destructive to the URL as possible.

**Comment by m-scimonelli** [comment]
Using the OP as inspiration, I made a `ReactRouter6Adapter` alternative:

```
const ReactRouter6Adapter: QueryParamAdapterComponent = ({ children }) => {
  const navigate = useNavigate();
  const location = useLocation();

  const adapter: QueryParamAdapter = {
    replace(location) {
      navigate(location, { replace: true });
    },
    push(location) {
      navigate(location);
    },
    get location() {
      return location;
    },
  };

  return children(adapter);
};
```

Didn't have the chance to do a complete round of testing, but it seems to be working as intended.
